File: 백준/Silver/10845. 큐/큐.py
import sys

num = int(sys.stdin.readline())

queue = []
arr = [input() for _ in range(num)]

for s in arr:
    if "push" in s:
        a, b = s.split(" ")
        queue.append(b)

    elif s == "front":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])
    elif s == "back":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])
    elif s == "size":
        print(len(queue))
    elif s == "empty":
        if len(queue) != 0:
            print(0)
        else:
            print(1)
    elif s == "pop":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.pop(0))

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.LinkedList;

public class Main {

	public static void main(String[] args) throws Exception {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());
		LinkedList<Integer> queue = new LinkedList<>();
		
		for(int i = 0; i < n;i++) {
			String[] split = br.readLine().split(" ");

			switch (split[0]) {
				case "pop":
					System.out.println(queue.isEmpty() ? -1 : queue.poll());
					break;
				
				case "front":
					System.out.println(queue.isEmpty() ? -1 : queue.peek());
					break;
					
				case "back":
					System.out.println(queue.isEmpty() ? -1 : queue.getLast());
					break;
				
				
				case "size":
					System.out.println(queue.size());
					break;
				
				case "empty":
					System.out.println(queue.isEmpty() ? 1 : 0);
					break;
				default:
					queue.offer(Integer.parseInt(split[1]));
					break;
			}
		}

	}

}
